### DISCUSSION TITLE
Fluvial-Aeolian Landforms

### DISCUSSION THREAD

**Discussion by maddykelley** [openearth/aeolis-python#219] [Case Studies]
![image](https://github.com/user-attachments/assets/f419edd8-caa0-4a18-8b66-0c1570c2ffb3)

Sedimentary environments, like river channels, floodplains, and beaches, frequently evolve through non-linear relationships due to spatial sediment transport gradients by winds, waves, and currents (Rhoads, 2013). It is common to approach geomorphic research from discrete units of study such as fluvial, glacial, or aeolian (Sherman, 1999), but long-term, net landscape changes often reflect the aggregate effects of numerous discrete processes. For example, transgressions and regressions of water levels (stages) on sandy landforms adjacent to water, such as coastal and lacustrine beaches, influence aeolian sediment transport (Ahmady-Birgani et al., 2018; Belnap et al., 2011; Draut, 2012; Licht et al., 2016; Sweeney et al., 2019) (Figure 2). Transgressions may specifically cause aeolian sediment transport to be supply-limited either through direct inundation or increases in interparticle bonding and sediment cohesion, making sediment less mobile (Belly, 1964; Chepil, 1956; McKenna-Neuman & Nickling, 1989; Sherman et al., 1998). Thus, transgressive and regressive events provide an important opportunity to study how hydrodynamic and aeolian processes interact, including along the sandy banks of fluvial systems where dune systems can be common morphologic features.

In river corridors, like the Colorado River within GCNP, the transfer of sediment occurs across various landforms within and outside of the active channel. Disturbance to the river or management activities can impact sediment transfer. These disturbances include shifts in land use (Walling & Fang, 2003; Kasprak et al., 2013; Wohl, 2020) and altered hydro-climatic regimes (Poesen et al., 2003; Christensen et al., 2004; Palmer et al., 2008; Belmont et al., 2011). Additional impacts on sediment connectivity can also include river diversion or irrigation works that modify flow regimes (Gaeuman et al., 2005; Mossa, 2016; Wohl, 2016; Jones, 2018) and the operation of dams (Schmidt & Wilcock, 2008; Grant et al., 2013; Magilligan et al., 2016). 

Along the Colorado River, there are numerous spatially co-occurring river sandbars and aeolian dunes, with the sandbars serving as the source of windblown sand supply for the adjacent and downwind aeolian dunes (Draut, 2012) (Figure 2). These fluvial-aeolian landforms are the key interest of study for this project. In addition to wind strength, aeolian sediment transport is influenced by various bed surface properties like grain size, moisture content, surface crusts, bed slopes, vegetation, non-erodible (roughness) elements, and anthropogenic disturbances. Previous work on these features has explored quantifying river regulation by dams and aeolian landscapes (Draut, 2012), the role of drying on the aeolian transport of river-sourced sand (Sankey et al., 2022), river flooding and sand supply for aeolian dune landscapes (Sankey et al., 2018), and past and future river hydrology and sediment availability for aeolian transport (Kasprak et al., 2021). 

This case study applies the (SUPER DUPER AWESOME!) AeoLiS model (Hoonhout & de Vries & AND ALL OTHER AMAZING FUNNY PEOPLE, 2016) to this fluvial-aeolian landscape and explore the supply-limiting scenarios. Given the many overlaps in relevant processes and types of landforms present in coastal-aeolian and fluvial-aeolian systems, this project aims to adapt and implement the existing coastal numerical morphodynamic model to quantify behavior across the land-water interface along the banks of the river through GCNP. The project objective is to develop and implement novel model workflows to predict the aeolian erosion, transport, or deposition of sediment from subaerially exposed river-sourced sediment deposits as a function of variability in river discharge related to changing water supply, use, and regulation. This work will advance scientific understanding of aeolian-fluvial interactions, advance technical capabilities to quantify non-coastal aeolian transport dynamics, and provide actionable guidance to land managers in these systems.

**Comment by maddykelley** [comment]
Update!

I've had some success. Here is the ROI I'm using for the grid:
![Figure_3_2019](https://github.com/user-attachments/assets/dfd8581e-953e-4a64-a535-41bcd547db3e)
![Figure_1_2019](https://github.com/user-attachments/assets/8a16b9e5-8b93-40db-9619-75c09e05c247)

I'm using a constant onshore windspeed of 10 m/s, a small tidal range (-0.2 to 0.2), and no waves. I have a `nb` layer 0.5 m below the surface.

I'm able to see some cool (and somewhat expected) transport, erosion, and deposition running it for 1 month.

![Results](https://github.com/user-attachments/assets/cc650609-f8f8-4337-8216-ce6d343766db)
![Results2](https://github.com/user-attachments/assets/f2d27ca5-3dab-46f7-b386-73f34d266f2c)

But this only works when I use `duran` (not `duran_full`). When I use `duran_full`, I get errors like this:

> C:\Users\mmkelley\Documents\AEOLIS\Versions\aeolis-python-main\aeolis\model.py:2468: RuntimeWarning: overflow encountered in multiply
>   pickup_i = (w_i * Cu_i - Ct_i) / Ts_i * self.dt # Dit klopt niet! enkel geldig bij backward euler
> 015.1%       0:01:30 /    0:09:57 /        0:08:27 / 3600.0
> C:\Users\mmkelley\AppData\Local\miniforge3\envs\env_aeolis_oregon\lib\site-packages\scipy\optimize\_zeros_py.py:438: RuntimeWarning: invalid value encountered in greater_equal
>   p1 = p * (1 + dx) + np.where(p >= 0, dx, -dx)
> C:\Users\mmkelley\Documents\AEOLIS\Versions\aeolis-python-main\aeolis\transport.py:204: RuntimeWarning: invalid value encountered in divide
>   return (veff_i - u_i) * np.abs(veff_i - u_i) / (uf_i ** 2) - u_i / (2 * alpha_i * np.abs(u_i)) - dh_i
> C:\Users\mmkelley\AppData\Local\miniforge3\envs\env_aeolis_oregon\lib\site-packages\scipy\optimize\_zeros_py.py:449: RuntimeWarning: invalid value encountered in divide
>   dp = (q1 * (p1 - p))[nz_der] / (q1 - q0)[nz_der]
> dgstrf info 1
> C:\Users\mmkelley\Documents\AEOLIS\Versions\aeolis-python-main\aeolis\model.py:2441: MatrixRankWarning: Matrix is exactly singular
>   Ct_i += scipy.sparse.linalg.spsolve(A, yCt_i.flatten())
> dgstrf info 1

Note I'm using version 3.0.0rc3. Also, when using duran, I still get odd stripes throughout the shear/grain velocity estimates see below:

![figure_grid_initialization](https://github.com/user-attachments/assets/c063d5a1-12d6-436c-b3a1-4396af9eb701)
![figure_params_initialization](https://github.com/user-attachments/assets/aaa74ba1-bf1b-49b4-956c-9d90513ab193)
![figure_timeseries_initialization](https://github.com/user-attachments/assets/03dadc6e-6f4b-4628-af05-64d3213702f7)

I'd appreciate your thoughts! Happy to share my files.

**Reply by Sierd** [reply]
Hi Maddy, these results look pretty neat already. 

The Duran formulations for grain speed you are playing with are quite complex. We have adopted these formulations when developing the complex 3D cases such as the Barchan and Parabolic dunes. The grain speeds in these complex cases are dependent on morphology. The feedbacks between morphology and wind (and associated grain speed) are manifested in the shape of the dunes. Bear in mind, that these examples involve pretty long timescales. In your case I would start simpler formulation for grain speed. You can add complexity at the point you need it.

I would use "grainspeed = windspeed " for now.

Also, you may want to turn of the shear solver to generate some initial results. Turning off the shear solver simplifies the wind signal to a spatially non varying. The results that you will generate with this will be easier to interpret and you can add complexity later. 

The first complexity I would add myself is a temporally varying wind condition and a longer simulation time.  If it is needed to add further complexity if would take it step by step.

Hope this helps!

**Reply by maddykelley** [reply]
Thank you! I am changing the grainspeed method, shear solver, and the wind file and will post an update!

I've been thinking about the wind data. I'd like to better understand how the wind magnitude (and direction) is used. For example, my data comes from a meteorological station across the river nearby (~3m off the ground). For other case studies, do people use wind data from a met station in the back dune area (~10m AGL) or maybe buoys? Do others make manually adjustments to their data via the wind files? Can this be done in AeoLiS? In the future, should I consider adjusting the input wind data magnitude given its collection location? I believe I need to eventually adjust the directions because of my grid's orientation (5 degrees). I didn't see this mentioned in the read the docs. Sorry if I missed it!

**Reply by Sierd** [reply]
Regarding the wind data AeoLiS has different options. The simplest one is you assume spatially non varying wind. I always start with this one myself and I only introduce complexity when I need it to explain behavior and/or observations. 

But then the question remains, what would be a representative assumption in the case of wind time-series. AeoLiS uses an assumption for 'measurement height of the wind velocity - z[m]' which is set to 10m by default (see https://aeolis.readthedocs.io/en/latest/user/defaults.html). You may want to specify z = 3 in your input. This may also help with your question below.

**Comment by maddykelley** [comment]
Howdy!

From Sierd and Nick's advice I have made the following changes.
grainspeed = windspeed
process_shear = F
I'm using modified wind data from the met station. the data I have is from 2019. I modified/simplified the wind direction to be either 0 or 180 (on or offshore) based on a simple range. But the magnitudes are the same as the real data. I repeated the data so I can model change (and compared to fielddata) across two years.

Measured change (from tls data 2019 - 2021). _What I am hoping to see in my results! :)_
![Diff_both2d](https://github.com/user-attachments/assets/19dcc2f6-dd4f-4889-8e2b-440298a0524b)
![Diff_both1d](https://github.com/user-attachments/assets/f6def55b-0917-4e11-80d5-a4a15b1240d4)

AeoLiS file

> %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
> %% AeoLiS Basalt Post-Sprint Configuration  2-year                  %%
> %% Date: 2024-8-21                                                  %%
> %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
> 
> %%% File names
>        	xgrid_file = xs.txt
>        	ygrid_file = ys.txt
> 		bed_file = zb.txt
> 		wave_file = wave.txt
>         wind_file = wind.txt
>         tide_file = tidal.txt
> 		ne_file = nb.txt
> 
> %%% Time variables	  
> 	dt = 3600 %1hour
> 	tstart = 0
> 	tstop = 63072000 %365 days
> 
> %%% Grain size distribution and layers			 
>        	grain_dist = 1
>       	grain_size = 0.0002 
> 		layer_thickness = 0.1
> 		nlayers = 1
> 		method_grainspeed= windspeed
> 
> %%% Numerical variables
>         scheme = euler_backward
> 		solver = pieter
> 		process_avalanche = T
> 		wind_convention = cartesian
> 		boundary_onshore = gradient
> 		boundary_offshore = constant
> 		process_shear = F
> 		process_runup = F
> 		k = 0.001
> 		process_tide = T
> 		process_wet_bed_reset = T
> 		th_moisture = T
> 		process_moist = T
> 		th_roughness = F
> 		th_nelayer = T
> 
> %%% Output variables
> 		output_times = 86400 %1day
> 		output_types = avg
> 		output_vars = zb w w_bed uw uws mass pickup q qs uth Cu tau 
> 		visualization = T

Other parameters
![figure_grid_initialization](https://github.com/user-attachments/assets/d37b9293-7abb-4030-931f-796767c35f96)
![figure_params_initialization](https://github.com/user-attachments/assets/2108ed9f-b5e9-49cb-9ef0-f428c9a844b3)
![figure_timeseries_initialization](https://github.com/user-attachments/assets/7e7af4b8-810a-4e79-9878-0d989f8bbcd7)



I am not getting much/any change :(
(Results from ncfile)
![results1](https://github.com/user-attachments/assets/bb1bef3c-07e8-4137-95a6-ad767db0b3f3)
![results2](https://github.com/user-attachments/assets/a4f8270c-e405-4016-8d94-6693def510d4)

I gut tells me this is because of wind magnitudes being low a lot of the time, but I did expect more changes...
Would turning on process_shear help? Other changes to explore before that?

**Reply by Sierd** [reply]
Hmmmmmmmmmmmm... My gut tells me the same as yours. Wind speeds may indeed be a bit low. I also see your earlier question about this. I will answer that in your post above.

Regarding the model I would do 2 things:
1. Verify your gut feeling by artifically increasing your wind speed in your time series. Just double it and see what happens. Results may not be physically relevant at this stage but you will learn how the model responds to the assumed wind speeds.
2. Pick a point where you know there is transport and plot timeseries of transport variables (q, Ct, Cu). You can learn a lot from that. If Ct=Cu than transport capacity if reached. If Ct<Cu than you have supply limited aeolian transport. q should be Ct times u_w. etc.
    Transport capacity (Cu) will be calculated using Bagnolds equation:
    https://aeolis.readthedocs.io/en/latest/user/model.html#equation-equilibrium-conc
    Note that you can choose other equations as well and there is some opportunity for calibration.

I am curious what you will find.

**Comment by bartvanwesten** [comment]
Hi maddy,

Using this conversation to react to your issue related to shear stresses (issue #227).
I have made some changes to your setup that seem to partially resolve some of the issues: 
[maddy_setup.zip](https://github.com/user-attachments/files/16905196/maddy_setup.zip)

In general, it seems that the shear solver has some difficulties with your topography. The instabilities mainly occur around the boundaries of your domain and propagate inwards. If you could try and increase the spatial extend of the domain, beyond the dune into more flat areas that could probably help.

For now, I made the following changes:
1. Used the waterlevel (zs) instead of the bedlevel (zb) for the shear calculation (see changed lines 225-227 in wind.py). This seems to resolve some of the offshore wiggles in shear stress.
2. Modified some other shear related parameters, to reduce the instabilities a bit further (see aeolis.txt).
3. Added a conservative value for max_bedlevel_change. This parameter reduces the timestep based on the amount of maximum bed level change in the model. This partially prevents the instabilities becoming larger over time.

In addition, I took the liberty to change some of the vegetation settings as well.
1. Increased vertical growth rate (V_ver), to make sure vegetation gets not buried to quickly
2. Non-vegetated cells now have a rhoveg of 0 instead of 0.001 (not vegetated) and vegetated cells of 1 instead of 2  (maximum value for density)
3. Set veg_min_elevation to 4, so no vegetation will grow below this elevation.

With these settings, your dense vegetation should remain in place, while allowing new vegetation to start growing in the bare cells.

Hopefully this helps a bit, good luck!

**Reply by maddykelley** [reply]
Bart! I greatly appreciate you looking into this! Let me take a look at everything and get back to you.
Many thanks again.
Maddy
